packages/fields-wysiwyg-tinymce/README.md

```diff
@@ -0,0 +1,32 @@
+---
+section: field-types
+title: Wysiwyg
+---
+
+# WYSIWYG HTML Field powered by TinyMCE Editor
+
+This field inserts a string path into your schema based on the `Text` field type implementation, and renders a WYSIWYG editor in the Admin UI using [TinyMCE](https://www.tiny.cloud/)
+
+## Usage
+
+In addition to using this field type in a List, for it to work in the Admin UI you'll need to serve the TinyMCE assets from your webserver.
+
+The Field includes a function that sets up the middleware for you. Configure it in your custom server like this:
+
+```js
+const keystone = require('@keystone-alpha/core');
+const WysiwygField = require('@keystone-alpha/fields-wysiwyg-tinymce');
+
+keystone.prepare().then(async ({ server }) => {
+  WysiwygField.bindStaticMiddleware(server);
```

Comment:
Why is this necessary? It feels really annoying to have to do this because we're breaking out of the API of field types which people are used to.

Reply:
I'm ok with this. It's not quite a _field_, but it's a part that's necessary for the field to work. Really, fields can export whatever they like, as long as their main export includes the minimum stuff for KS5 to understand it. If it exports more; so be it 👍

Reply:
So this was really tricky because TinyMCE basically *needs* to be able to load files from its package at runtime. We either have to get users to sign up for a `tiny.cloud` account and provide an API key, or we serve them ourselves from _somewhere_.

Basically you just can't vendor this stuff from an npm module using webpack, there are just too many issues that are fundamental to TinyMCE's architecture and history that prevent that from being possible - at least as far as they've got in v5. I spent _hours_ trying not to need this and tl;dr failed.

(to be clear; their official recommendation in the docs is to _literally copy the files when you download the package, into somewhere in your repo that static assets are hosted_. This is not something I want to do, so consider the solution here a middleground.)

Now that the field is decoupled from keystone, the admin ui, and the webserver, it's harder than it was in Keystone 4's monolith (in a good way) to serve random static resources. I was actually really happy with this solution, because it's a neat solution to an otherwise horrid problem.

If the field package exports its own bindings, and they're easy to plug into the server, it means everything is still encapsulated and we don't have issues with updating tinymce / etc like we have in the past.

Reply:
Could you add a comment basically saying what you said here for people looking at this in the future?

Reply:
My concept in #842  would help here.  basically postponing field initialization and server preparation to the very end of processing queue. This way we can have `keystone`/`server` instance passed to Field type at the initialization, which would remove all this complexity.

We have to pass keystone/server to `list.initFields()`